src/Components/MediaChannelComponent.ts

```diff
@@ -0,0 +1,176 @@
+            await message.channel.send('Command syntax: \`setmc #channel1 #channel2 #channel3 ...\`');
+            await message.channel.send('You can also use \`getmc\` to see a list of set channels.');
+        }
+        // TODO: also accept channels by name - make getGuildChannels accept channel names as well?
+        else {
+            for (const arg of args) {
+                let chid = arg.replace(/(\<|\>|\#)/gi, "");
+                const channel = this.djmtGuild.getGuildChannel(chid) as TextChannel;
+                if (!channel) {
+                    await message.channel.send(`${arg} is not a valid channel`);
+                }
+                else {
+                    // find channel in the list of media channels, if it exists
+                    let exists = false;
+                    this.channelsArray.forEach( (item, index) => {
+                        if(item.id === chid) {
+                            exists = true;
+                            // really, js? why is this a thing
+                            this.channelsArray.splice(index,1);
+                        }
+                    });
+                    if (exists) {
+                        await message.channel.send(`Removed ${arg} as a media channel.`);
+                    }
+                    else {
+                        this.channelsArray.push(channel);
+                        await message.channel.send(`Successfully added ${arg} as a media channel.`);
+                    }
+                }
+            }
+        }
+        await this.djmtGuild.saveJSON();
+    }
+
+    /**
+     * Returns a list of media channels.
+     * @param message The message object of the command message.
+     * @private
+     */
+    private async getMediaChannel(message: Message): Promise<void> {
+        if (!isAdmin(message)) {
+            await message.channel.send(`This command requires administrator permissions.`);
+            return;
+        }
+        if (this.channelsArray.length === 0) {
+            await message.channel.send(`There are no set media channels.`);
+            return;
+        }
+        let m = 'The current set media channels are:\n';
+        for (const c of this.channelsArray) {
+            m += `<#${c.id}>\n`;
+        }
+        await message.channel.send(m);
+    }
+
+    /**
+     * Checks and deletes a message if it's in a media channel and doesn't have media.
+     * @param message The message to be checked.
+     * @private
+     */
+    private async checkMedia(message: Message): Promise<void> {
+        if (this.channelsArray.indexOf(<TextChannel>message.channel) > -1) {
+            // delete all messages without attachments or links
+            if (message.attachments.size === 0 && !(this.linkRegex.test(message.content))) {
+                let msg = `${message.author.toString()}, your message has been deleted because it does not have media.`;
+                await message.delete();
```

Comment:
The only bug i found here was that if the message is deleted by the time `await message.delete()` is called, it causes Unhandled Promise Rejection 
i got this to happen by sending a message with djmt!say, since that function tries to delete the original message as soon as it's sent, so it's a pretty rare case. A try catch is probably all it needs.